api/v1alpha1/managedfleetnotificationrecord_types_test.go

```diff
@@ -0,0 +1,153 @@
+				nr := testMNFR.Status.NotificationRecordByName[0]
+				_, err := testMNFR.UpdateNotificationRecordItem(nr.NotificationName, "nope")
+				Expect(err).NotTo(BeNil())
+			})
+		})
+	})
+
+	Context("When adding a notification record item", func() {
+		Context("If the notification record item doesn't exists", func() {
+			It("will add the item", func() {
+				nr := testMNFR.Status.NotificationRecordByName[0]
+				nfi, err := testMNFR.AddNotificationRecordItem("test-hc-1-4", &nr)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(testMNFR.Status.NotificationRecordByName[0].NotificationRecordItems)).To(Equal(4))
+				Expect(nfi.HostedClusterID).To(Equal("test-hc-1-4"))
+			})
+		})
+		Context("If the notification record item already exists", func() {
+			It("will error", func() {
+				nr := testMNFR.Status.NotificationRecordByName[0]
+				nfi, err := testMNFR.AddNotificationRecordItem("test-hc-1-3", &nr)
+				Expect(err).To(HaveOccurred())
+				Expect(nfi).To(BeNil())
+			})
+		})
+		Context("When the notification does not exist", func() {
+			It("will return an error", func() {
+				_, err := testMNFR.UpdateNotificationRecordItem("nope", "nope")
+				Expect(err).NotTo(BeNil())
+			})
+		})
+		Context("When the notification record item does not exist", func() {
+			It("will return an error", func() {
+				nr := testMNFR.Status.NotificationRecordByName[0]
+				_, err := testMNFR.UpdateNotificationRecordItem(nr.NotificationName, "nope")
+				Expect(err).NotTo(BeNil())
+			})
+		})
+	})
+
+	Context("When removing a notification record item", func() {
+		Context("When the notification record item already exists", func() {
+			It("will update it correctly", func() {
+				nr := testMNFR.Status.NotificationRecordByName[0]
+				nri := testMNFR.Status.NotificationRecordByName[0].NotificationRecordItems[1]
+				nr2, err := testMNFR.RemoveNotificationRecordItem(nr.NotificationName, nri.HostedClusterID)
+				Expect(err).To(BeNil())
+				Expect(len(testMNFR.Status.NotificationRecordByName[0].NotificationRecordItems)).To(Equal(2))
+				Expect(testMNFR.Status.NotificationRecordByName[0].NotificationRecordItems[0].HostedClusterID).To(Equal("test-hc-1-1"))
+				Expect(testMNFR.Status.NotificationRecordByName[0].NotificationRecordItems[1].HostedClusterID).To(Equal("test-hc-1-3"))
+				Expect(nr2.NotificationRecordItems[0].HostedClusterID).To(Equal("test-hc-1-1"))
+				Expect(nr2.NotificationRecordItems[1].HostedClusterID).To(Equal("test-hc-1-3"))
+			})
+		})
+		Context("When the notification does not exist", func() {
+			It("will return an error", func() {
+				_, err := testMNFR.RemoveNotificationRecordItem("nope", "nope")
+				Expect(err).NotTo(BeNil())
+			})
+		})
+		Context("When the notification record item does not exist", func() {
+			It("will return an error", func() {
+				nr := testMNFR.Status.NotificationRecordByName[0]
+				_, err := testMNFR.UpdateNotificationRecordItem(nr.NotificationName, "nope")
+				Expect(err).NotTo(BeNil())
+			})
+		})
+	})
```

Comment:
Similar query for this block as well if the Update test intended to be under remove context block?

Reply:
Fixed